### PR TITLE
Add support for `approle` auth method

### PR DESCRIPTION
**Pull request by cognifloyd**
This adds an approle authentication method in addition to the default
token-based auth method. It also adds configuration to select which
auth_method to use. The implementation of auth_method should make it
straight forward to add other autgh methods in the future.

Plus, this bumps the hvac min version to v0.10.6 (released 14 Dec 2020)
because the `client.auth.approle.login` method was added in that
release, deprecating the old `client.auth_approle` method.

### CODE REVIEW

**Review by nmaludy — CHANGES_REQUESTED** [review]
@cognifloyd looks like there is a lot of "noise" in this change, things like changing formatting. Can you please clean this up so we can get a more focused picture on what is actually going on.

Thanks,

**Comment by cognifloyd** [comment]
K. I have another couple of PRs and then we can bump the version.

File: actions/lib/action.py [hunk 1]
```diff
@@ -10,10 +10,29 @@ def __init__(self, config):
...
+            pass
+
+        # where implemented prefer: client.auth.<method>.login
+        # as many client.auth_* methods are deprecated.
+
+        elif auth_method == "approle":
+            client.auth.approle.login(
+                role_id=self.config["role_id"],
+                secret_id=self.config["secret_id"],
+            )
```
Comment (nmaludy) [inline actions/lib/action.py:35]: Probably good for a warning here if the auth_method doesn't exist.

File: actions/lib/action.py [hunk 2]
```diff
@@ -10,10 +10,29 @@ def __init__(self, config):
...
+
+        # token is passed during client init to allow client to also
+        # get the token from VAULT_TOKEN env var or ~/.vault-token
         client = hvac.Client(url=url, token=token, verify=verify)
+
+        # token is handled during client init
+        # other auth methods will override it as needed
+        if auth_method == "token":
+            pass
+
```
Comment (nmaludy) [inline actions/lib/action.py]: lots of extra space here makes it look like the `if` and `elif` aren't connected.

**Review by nmaludy — CHANGES_REQUESTED** [review]
Couple really small comments

File: actions/lib/action.py (hunk 1 above)
Comment (cognifloyd) [reply actions/lib/action.py:35]: I went with a `NotImplementedError`. You should only get this if you're manually interacting with this python code somehow. StackStorm itself enforces that auth_method is one of `token` or `approle` with the enum in action metadata.

**Comment by nmaludy** [comment]
@cognifloyd looks like build is failing now

**Comment by cognifloyd** [comment]
fixed

**Review by nmaludy — APPROVED** [review]

**Pull request merged**